alot/commands/__init__.py

```diff
@@ -15,25 +15,44 @@ class Command(object):
 
     """base class for commands"""
     repeatable = False
+    mode = ""
+    cmdname = ""
 
     def __init__(self):
         self.undoable = False
         self.help = self.__doc__
 
-        class_ = type(self)
-        class_name = class_.__name__
-        cmdname, mode = reverse_lookup_command(class_)
-
-        # fetch and set pre and post command hooks
-        # they are set to None if not defined in the hooks file
-        get_hook = settings.get_hook
-        self.prehook = (get_hook('pre_%s_%s' % (mode, cmdname)) or
-                        get_hook('pre_global_%s' % cmdname))
-        self.posthook = (get_hook('post_%s_%s' % (mode, cmdname)) or
-                         get_hook('post_global_%s' % cmdname))
+        # set to False if no attemtp to fetch hooks from settings
+        self._prehook = False 
+        self._posthook = False
 
-        logging.debug("%s prehook: %s", class_name, str(self.prehook))
-        logging.debug("%s posthook: %s", class_name, str(self.posthook))
+    @property
+    def id(self):
+        """Mode and command name of instance as string."""
+        class_ = type(self)
+        return class_.mode + '.' + class_.cmdname
+
+    @property
+    def prehook(self):
+        """Command pre hook."""
+        if self._prehook == False:
```

Comment:
Two things:
1) Don't use `==` with singletons (`None`, `True`, `False`, and `NotImplemented` are the most common), use `is` (and use `is not` instead of `!=`), It's faster
2) You can just write this as `if not self._prehook`